modules/gateway/peers.go

```diff
@@ -162,6 +163,40 @@ func (g *Gateway) acceptConn(conn net.Conn) {
 		return
 	}
 
+	// Read the peer's port that we can dial them back on. Peers older than v0.6.1
+	// will only be discovered by newer peers via the ShareNodes RPC.
+	var remoteAddr modules.NetAddress
+	if build.VersionCmp(remoteVersion, "0.6.1") >= 0 {
+		var dialbackPort string
+		err := encoding.ReadObject(conn, &dialbackPort, 13) // Max port # is 65535 (5 digits long) + 8 byte string length prefix
+		if err != nil {
+			conn.Close()
+			g.log.Printf("INFO: could not read remote peer's (%v) port: %v", addr, err)
+			return
+		}
+		if _, err := strconv.Atoi(dialbackPort); err != nil {
+			conn.Close()
+			g.log.Printf("INFO: peer (%v) sent an invalid dialback port: %v", addr, err)
```

Comment:
To me, purpose of logging is to give the user a clean history of the things that their client has done. Stuff like 'started up, made 5 contracts, renewed on X date for Y coins', etc. Cluttering it with a bunch of debugging statements is counter productive unless you are doing debugging, in which case `Debugln` will give you what you want.

If we let byzantine users control production machine logging, they can fill users disks in a DoS attack. Definitely not a good thing.